src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,145 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { useRef } from 'react';
+import { NumberInputIncrement, NumberInputProps } from './types';
+import { merge } from '@utilities/merge';
+import { FOCUS_STYLE } from '@utilities/focusStyle';
+import { useFocusRing } from '@react-aria/focus';
+import { IconExclamationMarkTriangle16, IconMinus16, IconPlus16 } from '@foundation/Icon';
+import { HelperText, HelperTextStyle } from '@utilities/input';
+import { useMemoizedId } from '@hooks/useMemoizedId';
+import { InputTypes } from 'src/types/input';
+
+export const NumberInput = ({
+    id: propId,
+    size,
+    error,
+    disabled = false,
+    required,
+    readOnly,
+    errorText,
+    decorator,
+    incrementable,
+    onChange,
+    onKeyDown,
+    onBlur,
+    onFocus,
+    'data-test-id': dataTestId = 'fondue-number-input',
+    ...props
+}: NumberInputProps) => {
+    const inputElement = useRef<HTMLInputElement | null>(null);
+    const numberInputType: InputTypes = 'number';
+    const buttonInputType: InputTypes = 'button';
+    const { isFocusVisible, focusProps } = useFocusRing({ within: true, isTextInput: true });
+
+    const handleOnChange = (value: string) => {
+        onChange?.(value);
+    };
+
+    const handleCount = (value: string, type?: NumberInputIncrement) => {
+        let newValue = Number(value) || 0;
+        switch (type) {
+            case NumberInputIncrement.DECREMENT:
+                newValue -= 1;
+                break;
+            case NumberInputIncrement.INCREMENT:
+                newValue += 1;
+                break;
+            default:
+                newValue = Number(value);
+        }
+        if (inputElement.current) {
+            inputElement.current.value = newValue.toString();
+            handleOnChange(inputElement.current.value);
+        }
+    };
+
+    return (
+        <div data-test-id={`${dataTestId}-component`}>
+            <div
+                {...focusProps}
+                className={merge([
+                    'tw-flex tw-items-center tw-h-9 tw-gap-2 tw-px-3 tw-border tw-transition tw-rounded tw-text-s tw-font-sans tw-relative tw-bg-white dark:tw-bg-transparent',
+                    'tw-border-solid',
+                    'focus-within:tw-border-black-90 hover:tw-border-black-90',
```

Comment:
No use of brand-tokens please, Only alias-tokens. according to Figma it should be `-line-xx-strong`.
Also it wouldn't hurt to have all border classes in the same line.

Reply:
Actually reviewing Design, the border styling is wrong.
It should be 
default state: `line-strong`
hover: `line-x-strong`
focus: `line-xx-strong`